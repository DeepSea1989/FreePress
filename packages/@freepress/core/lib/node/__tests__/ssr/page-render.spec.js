const fs = require('fs-extra');
const { getSourceDirs } = require('@freepress/test-util');
const { Helmet } = require('react-helmet');
const App = require('../../app');

jest.setTimeout(200000);
// Helmet.canUseDOM = false;
describe('ssr', () => {
  beforeEach(() => {
    Helmet.canUseDOM = false;
  });

  afterEach(() => {
    Helmet.canUseDOM = true;
  });
  test('pagerender', async () => {
    const getPageRenderer = require('../../app/getPageRenderer');
    await Promise.all(
      getSourceDirs(__dirname).map(async ({ name, docsPath, docsTempPath }) => {
        await fs.ensureDir(docsTempPath);
        const app = await new App(docsPath).process(true, true);
        const { clientWebpackConfig, serverWebpackConfig, options } = app;
        const pagerender = await getPageRenderer(serverWebpackConfig, clientWebpackConfig, options);
        const html = await pagerender.renderPage(
          options.siteData.pages[0],
          true /* only render html*/
        );
        expect(html).toMatchSnapshot();
      })
    );
  });
});
