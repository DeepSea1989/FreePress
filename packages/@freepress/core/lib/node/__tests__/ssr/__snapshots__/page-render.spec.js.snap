// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ssr pagerender 1`] = `
"<!-- prettier-ignore -->
<!DOCTYPE html>
<html {{{ htmlAttr }}}>
  <head>
    <meta charset=\\"utf-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width,initial-scale=1\\" />
     <!--react-helmet render area  -->
    {{{ title }}}
     {{{ meta }}}
     {{{ helmet-links }}}
    <script>
     window.RC_CONTEXT = \\"{{ RC_CONTEXT }}\\";
    </script>
    
    <!--loadable-components prefetch  -->
    {{{ links }}} 
    {{{ style }}}  
  </head>
  <body>
    <div id=\\"app\\"><div class=\\"page-wrapper index-page-wrapper\\"><div style=\\"width:100%\\"><div class=\\"\\"><div id=\\"header\\" class=\\"header\\"><div class=\\"ant-row\\"><div class=\\"ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-8 ant-col-lg-8 ant-col-xl-5 ant-col-xxl-4\\"><a id=\\"site-logo\\" href=\\"/\\"><span class=\\"left-top-title\\">freepress</span></a></div><div class=\\"ant-col ant-col-xs-0 ant-col-sm-0 ant-col-md-16 ant-col-lg-16 ant-col-xl-19 ant-col-xxl-20\\"><div class=\\"header-meta\\"><div class=\\"right-header\\"></div><div id=\\"menu\\"><ul id=\\"nav\\" class=\\"ant-menu ant-menu-light ant-menu-root ant-menu-horizontal\\" role=\\"menu\\"></ul></div></div></div></div></div></div></div><div class=\\"ant-row main-wrapper\\"><div class=\\"ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-24 ant-col-xl-24 ant-col-xxl-24\\"><div><div><div class=\\"main-container\\"><article class=\\"markdown\\"><div class=\\"modifiedTime modifiedTimeLeft\\">Last Updated<!-- --> <!-- -->2020-05-09 10:48:00</div><div class=\\" toc-affix\\"><div><div class=\\"\\"><div class=\\"toc toc-affix ant-anchor-wrapper\\" style=\\"max-height:calc(100vh - 70px)\\"><div class=\\"ant-anchor\\"><div class=\\"ant-anchor-ink\\"><span class=\\"ant-anchor-ink-ball\\"></span></div><div class=\\"ant-anchor-link\\"><a class=\\"ant-anchor-link-title\\" href=\\"#hello-world\\" title=\\"hello world!\\">hello world!</a></div></div></div></div></div></div><section class=\\"markdown api-container\\"><h1 class=\\"__internal\\" id=\\"hello-world\\">hello world!<a href=\\"#hello-world\\" aria-hidden=\\"true\\" class=\\"anchor\\">#</a></h1><p>hello world</p></section><section class=\\"prev-next-nav\\"></section></article></div></div></div></div></div></div></div>
     <!--loadable-components render area  -->
    {{{ scripts }}}
  </body>
</html>
"
`;
